code/modules/mob/living/simple_animal/hostile/headcrab.dm

```diff
@@ -0,0 +1,79 @@
+#define EGG_INCUBATION_TIME 180
+
+/mob/living/simple_animal/hostile/headcrab
+	name = "Headslug"
+	desc = "Absolutely not de-beaked or harmless. Keep away from corpses."
+	icon_state = "headcrab"
+	icon_living = "headcrab"
+	icon_dead = "headcrab_dead"
+	health = 20
+	maxHealth = 20
+	melee_damage_lower = 5
+	melee_damage_upper = 5
+	attacktext = "chomps"
+	attack_sound = 'sound/weapons/bite.ogg'
+	faction = list("creature")
+	robust_searching = 1
+	stat_attack = 2
+	environment_smash = 0
+	var/datum/mind/origin
+
+/mob/living/simple_animal/hostile/headcrab/proc/Infect(var/mob/living/carbon/human/victim)
+	var/obj/item/changeling_egg/egg = new(victim)
+	egg.owner = origin
+	victim.internal_organs += egg
+	visible_message("<span class='notice'>[src] lays an egg in a [victim]!</span>")
+
+/mob/living/simple_animal/hostile/headcrab/AttackingTarget()
+	if(ishuman(target))
+		var/mob/living/carbon/human/H = target
+		if(H.stat == DEAD)
+			Infect(target)
+			src.Die()
+			return
+	target.attack_animal(src)
+
+/obj/item/changeling_egg
```

Comment:
This is duplicated from the larva code. Maybe these two classes can share an abstract class to remove the duplicated code.